src/routes/accounts.py

```
@@ -17,9 +17,289 @@
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid email or password."
+        )
+    if not user.is_active:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="User account is not activated."
+        )
+    access_token = jwt_manager.create_access_token(
+        data={"user_id": user.id, "email": user.email}
+    )
+    refresh_token = jwt_manager.create_refresh_token(
+        data={"user_id": user.id, "email": user.email},
+        expires_delta=timedelta(days=settings.LOGIN_TIME_DAYS)
+    )
+    try:
+        refresh_token_obj = RefreshTokenModel.create(
+            user_id=user.id,
+            days_valid=settings.LOGIN_TIME_DAYS,
+            token=refresh_token
+        )
+        db.add(refresh_token_obj)
+        await db.commit()
+        return UserLoginResponseSchema(
+            access_token=access_token,
+            refresh_token=refresh_token,
+            token_type="bearer"
+        )
+    except SQLAlchemyError:
+        await db.rollback()
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="An error occurred while processing the request."
+        )
+
+
+@router.post(
+    "/refresh/",
+    response_model=TokenRefreshResponseSchema,
+    status_code=status.HTTP_200_OK,
+)
+async def refresh_access_token(
+    payload: TokenRefreshRequestSchema,
+    db: AsyncSession = Depends(get_db),
+    jwt_manager: JWTAuthManagerInterface = Depends(get_jwt_auth_manager),
+):
+    refresh_token = payload.refresh_token
+    if not refresh_token:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Refresh token is required."
+        )
+
+    try:
+        token_data = jwt_manager.decode_refresh_token(refresh_token)
+        user_id = token_data.get("user_id")
+    except Exception:
```

The refresh token decode uses a broad `except Exception:`. Per the requirements/tests you should catch the specific exceptions the jwt_manager (or underlying JWT library) raises (e.g. `ExpiredSignatureError`, `InvalidTokenError`, or custom exceptions from your jwt_manager) and map them to the exact HTTP responses expected by tests (400 with message like "Token has expired."). Avoid a bare `except Exception:` so real errors aren't swallowed and the tests can assert the correct messages/status codes.